module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -5343,14 +5343,16 @@ zfs_getextattr_dir(struct vop_getextattr_args *ap, const char *attrname)
 	vp = nd.ni_vp;
 	NDFREE(&nd, NDF_ONLY_PNBUF);
 	if (error != 0)
-		return (error);
+		return (SET_ERROR(error));
 
 	if (ap->a_size != NULL) {
 		error = VOP_GETATTR(vp, &va, ap->a_cred);
 		if (error == 0)
 			*ap->a_size = (size_t)va.va_size;
 	} else if (ap->a_uio != NULL)
 		error = VOP_READ(vp, ap->a_uio, IO_UNIT, ap->a_cred);
+	if (error != 0)
+		error = SET_ERROR(error);
```

Comment:
Here and in few places below VOP_*() calls are really going back to ZFS through some debugging wrappers.  Does this really count as first error, considering SET_ERROR() likely was called there already?

Reply:
No I suppose not, I hadn't considered that. Thanks :)